src/test/java/webdriverFramework/SSMission.java

```diff
@@ -1,53 +1,211 @@
+            thirdSelectedAd = randInt(electronicsSearchResultPage.allAdvList.size());
+        }
 
-        driver.findElement(By.xpath("//*[@class=\"a1\"][@href=\"/ru/electronics/\"]")).click();
-        wait.until(ExpectedConditions.elementToBeClickable(By.xpath("//*[@href='/ru/electronics/search/']")));
-        driver.findElement(By.xpath("//*[@href='/ru/electronics/search/']")).click();
-        wait.until(ExpectedConditions.elementToBeClickable(By.id("ptxt")));
-        driver.findElement(By.xpath("//*[@id='ptxt']")).sendKeys("Лаптоп");
-        wait.until(ExpectedConditions.elementToBeClickable(By.xpath("//*[@id='cmp_1']")));
-        driver.findElement(By.xpath("//*[@id='cmp_1']")).click();
 
-//        Select dropDownPodrubrika = new Select(driver.findElement(By.name("'sid'")));
-//        dropDownPodrubrika.selectByValue("1");
+//        WebElement firstAd = electronicsSearchResultPage.allAdvList.get(firstSelectedAd);
+//        WebElement secondAd = electronicsSearchResultPage.allAdvList.get(secondSelectedAd);
+//        WebElement thirdAd = electronicsSearchResultPage.allAdvList.get(thirdSelectedAd);
 
 
+
+
+
+/*
+        List<WebElement> selectedAdsList = new ArrayList<>();
+        selectedAdsList.add(firstAd);
+        selectedAdsList.add(secondAd);
+        selectedAdsList.add(thirdAd);
+
+
+        String firstAdvText = electronicsSearchResultPage.getAdsText(firstAd);
+        String secondAdvText = electronicsSearchResultPage.getAdsText(secondAd);
+        String thirdAdvText = electronicsSearchResultPage.getAdsText(thirdAd);
+
+        Roma thinks this is not necessary anymore. 02/03/17 16:00.
+       */
+
+/*      Roma thinks this is not needed and can be replaced after, 02.03.2017 18:20
+
+        List<String> textsOfSelectedAds = new ArrayList<>();
+        textsOfSelectedAds.add(electronicsSearchResultPage.getAdAndItsText(selectedAdNumbers[0]));
+        textsOfSelectedAds.add(electronicsSearchResultPage.getAdAndItsText(selectedAdNumbers[1]));
+        textsOfSelectedAds.add(electronicsSearchResultPage.getAdAndItsText(selectedAdNumbers[2]));
+
+*/
+        electronicsSearchResultPage.createSelectedAdTextList(selectedAdNumbers); // replaces 4lines above
+
+
+
+
+
+
+
+//        textsOfSelectedAds.add(firstAd.findElement(By.xpath(electronicsSearchResultPage.PATHTOADSTEXT)).getText());
+//          this is not needed right now too
+
+
+
+        electronicsSearchResultPage.selectAdsCheckbox(firstAd);
+        electronicsSearchResultPage.selectAdsCheckbox(secondAd);
+        electronicsSearchResultPage.selectAdsCheckbox(thirdAd);
+
+        electronicsSearchResultPage.showSelectedAdsButton.click();
+
+        ElectronicsShowSelectedPage electronicsShowSelectedPage = new ElectronicsShowSelectedPage(driver);
+
+        List<String> textsOfSelectedFilteredAds = new ArrayList<>();
+        textsOfSelectedFilteredAds.add(electronicsShowSelectedPage.getSelectedAdvText(electronicsShowSelectedPage.selectedAds.get(0)));
```

Comment:
Why only 3 element, but not all ? Try to iterate via > electronicsShowSelectedPage.selectedAds and add all test to a list. Later, in case you have to find 4 elements - you will not need to do a code change.